index.js

```diff
@@ -33,22 +34,42 @@ module.exports.s3 = function(config, key, args, callback) {
 			Object.assign({ region: config.region }, s3config)
 		);
 	}
+
 	var s3 = regions[config.region];
-	var s3Request = authenticatedRequest ? s3.makeRequest : s3.makeUnauthenticatedRequest
-
-	return s3Request(
-		'getObject',
-		{ Bucket: config.bucket, Key: key },
-		function(err, data) {
-			if (err) {
-				return callback(err);
-			}
+	var isPresigned = !! args['X-Amz-Algorithm'];
+
+	if ( authenticatedRequest ) {
+		request = s3.makeRequest( 'getObject', { Bucket: config.bucket, Key: key } );
+	} else {
+		request = s3.makeUnauthenticatedRequest( 'getObject', { Bucket: config.bucket, Key: key } );
+		// To support forwarding presigned URLs, we hook into the post `build` step to add/forward
+		// the Amz signing URL query params from the current request.
+		if ( isPresigned ) {
+			request.addListener('build', function( req ) {
+				const params = querystring.stringify({
+					'X-Amz-Algorithm': args['X-Amz-Algorithm'],
+					'X-Amz-Content-Sha256': args['X-Amz-Content-Sha256'],
+					'X-Amz-Credential': args['X-Amz-Credential'],
+					'X-Amz-SignedHeaders': args['X-Amz-SignedHeaders'],
+					'X-Amz-Expires': args['X-Amz-Expires'],
+					'X-Amz-Signature': args['X-Amz-Signature'],
+					'X-Amz-Date': args['X-Amz-Date'],
+				});
+				req.httpRequest.path += `?${ params }`;
+			});
+		}
+	}
+	request.send( function(err, data) {
```

Comment:
```suggestion
	request.send( function( err, data ) {
```

Could we use an arrow function here btw?